source/acquaman/SGM/SGMLineScanController.cpp

```diff
@@ -22,14 +28,150 @@ SGMLineScanController::~SGMLineScanController()
+		foreach(CLSAmptekSDD123DetectorNew* amptekDetector, amptekDetectors_){
+			generalPurposeCounters.insert(amptekDetector, amptekDetector->retrieveGeneralPurposeCounterData(clientDataRequestMap_.value(amptekDetector->amdsBufferName())));
+			if(!highestAverageAmptekDetector || (generalPurposeCounters.value(amptekDetector).averageValue() > generalPurposeCounters.value(highestAverageAmptekDetector).averageValue()) )
+				highestAverageAmptekDetector = amptekDetector;
+		}
+
+		QList<QVector<qint32> > amptekBaseData;
+		amptekBaseData << generalPurposeCounters.value(highestAverageAmptekDetector).generalPurposeCounterVector();
+		amptekRangeData = highestAverageAmptekDetector->retrieveContinuousMotionRangeData(amptekBaseData, expectedDuration, 5);
+	}
+
+
+	int scalerInitiateMovementIndex = 0;
+	QList<QVector<qint32> > scalerBaseData;
+	scalerBaseData << scalerChannelRebaseVectors.value("HexapodRed") << scalerChannelRebaseVectors.value("HexapodBlack");
+	AMDetectorContinuousMotionRangeData scalerRangeData = scalerChannelDetectors_.first()->retrieveContinuousMotionRangeData(scalerBaseData, expectedDuration, 5);
+	if(scalerRangeData.isValid())
+		scalerInitiateMovementIndex = scalerRangeData.motionStartIndex();
+
+	qDebug() << "Amptek: " << amptekRangeData.motionStartIndex() << amptekRangeData.motionEndIndex() << amptekRangeData.listIndex();
+	qDebug() << "Hexapod Scaler: " << scalerRangeData.motionStartIndex() << scalerRangeData.motionEndIndex() << scalerRangeData.listIndex();
+
+	QVector<qint32> hexapodRedVector = scalerChannelRebaseVectors.value("HexapodRed");
+	// END OF STEP 4
+
+	// STEP 5: Generate the Axis Values "In Order"
+	QVector<double> globalXRecorderData = dataRecorderMap_.value("HexapodXRecorder");
+	QVector<double> globalYRecorderData = dataRecorderMap_.value("HexapodYRecorder");
+	QVector<double> globalZRecorderData = dataRecorderMap_.value("HexapodZRecorder");
+	QList<QVector3D> globalCoordinateRecorderPositions;
+	for(int x = 0, size = globalXRecorderData.count(); x < size; x++)
+		globalCoordinateRecorderPositions.append(QVector3D(globalXRecorderData.at(x), globalYRecorderData.at(x), globalZRecorderData.at(x)));
+
+	QList<QVector3D> primeCoordinateRecorderPositions = SGMBeamline::sgm()->hexapod()->transformVectors(globalCoordinateRecorderPositions);
+	int axisIndex = 0;
+	if(AMBeamline::bl()->exposedControlByInfo(configuration_->axisControlInfos().at(0)) == SGMBeamline::sgm()->hexapod()->yAxisPrimeControl())
+		axisIndex = 1;
+	else if(AMBeamline::bl()->exposedControlByInfo(configuration_->axisControlInfos().at(0)) == SGMBeamline::sgm()->hexapod()->zAxisPrimeControl())
+		axisIndex = 2;
+	// END OF STEP 5
+
+
+	// STEP 6: Place Data
+	CLSAMDSScalerChannelDetector *asScalerChannelDetector = 0;
+	AMDSLightWeightGenericFlatArrayDataHolder *dataHolderAsGenericFlatArrayDataHolder = 0;
+	for(int x = scalerInitiateMovementIndex, size = hexapodRedVector.count(); (x < size) && (x < 200); x++){
```

Comment:
I found that I needed to make the 200 a 200+scalerInitiateMovementIndex to get all the data to show up in the map scans.  Thoughts @davidChevrier?